packages/datadog-plugin-aws-sdk/src/util.js

```diff
@@ -84,8 +84,67 @@ const extractPrimaryKeys = (keyNames, keyValuePairs) => {
   }
 }
 
+/**
+ * Extracts queue metadata from an SQS queue URL for span tagging.
+ * Handles modern and legacy AWS endpoint formats, with or without schemes.
+ * Automatically detects AWS partitions (standard, China, GovCloud) from region.
+ *
+ * @param {string} queueURL - SQS queue URL in any supported format
+ * @returns {Object|null} Object with queueName and arn, or null if URL format is invalid
+ *
+ * @example
+ * // Modern AWS SQS URLs
+ * extractQueueMetadata('https://sqs.us-east-1.amazonaws.com/123456789012/my-queue')
+ * // Returns { queueName: 'my-queue', arn: 'arn:aws:sqs:us-east-1:123456789012:my-queue' }
+ *
+ * extractQueueMetadata('sqs.eu-west-1.amazonaws.com/123456789012/my-queue') // no scheme
+ * // Returns { queueName: 'my-queue', arn: 'arn:aws:sqs:eu-west-1:123456789012:my-queue' }
+ *
+ * // Legacy AWS SQS URLs
+ * extractQueueMetadata('https://us-west-2.queue.amazonaws.com/123456789012/legacy-queue')
+ * // Returns { queueName: 'legacy-queue', arn: 'arn:aws:sqs:us-west-2:123456789012:legacy-queue' }
+ *
+ * extractQueueMetadata('https://queue.amazonaws.com/123456789012/global-legacy-queue')
+ * // Returns { queueName: 'global-legacy-queue', arn: 'arn:aws:sqs:us-east-1:123456789012:global-legacy-queue' }
+ */
+const extractQueueMetadata = queueURL => {
+  if (!queueURL) {
+    return null
+  }
+
+  const parts = queueURL.split('/').filter(Boolean)
+
+  // Check if URL has scheme
+  const hasScheme = parts[0] && parts[0].startsWith('http')
+  const minParts = hasScheme ? 4 : 3
+
+  if (parts.length < minParts) return null
+
+  const accountId = parts[parts.length - 2]
+  const queueName = parts[parts.length - 1]
+  const host = hasScheme ? parts[1] : parts[0]
+
+  let region = 'us-east-1' // Default region if not found in URL
+  const hostParts = host.split('.')
+  if (host.includes('amazon') && hostParts.length > 3) {
+    // sqs.{region}.amazonaws.com or {region}.queue.amazonaws.com
+    region = hostParts[0] === 'sqs' ? hostParts[1] : hostParts[0]
```

Comment:
```suggestion
  if (host.includes('.amazonaws.com')) {
    // sqs.{region}.amazonaws.com or {region}.queue.amazonaws.com
    const startFrom = host.startsWith('sqs.') ? 4 : 0
    const nextDot = host.indexOf('.', startFrom)
    region = host.slice(startFrom, nextDot)
```

Just a suggestion. This prevents allocating a new array and needing to copy the string and is faster therefore. I do not know how frequently we use this method, since it's a fast operation already and is only beneficial in case it is frequently called.

Looking at the examples: could it actually ever happen that the host is not amazonaws.com?

Reply:
for all production use cases no, I'm just doing this check for the localstack test cases that are using localhost endpoints. I still wanted to test the arn generation logic within the integration tests, and `us-east-1` is the default region for localstack tests and legacy queues with no region specified